webapp/src/main/java/org/apache/atlas/notification/NotificationHookConsumer.java

```diff
@@ -477,6 +484,31 @@ public void doWork() {
             }
         }
 
+        private void setMessagePrincipal(String messageUser) throws AtlasException {
+            Configuration configuration = ApplicationProperties.get();
+            boolean msgPrincipalEnabled = configuration.getBoolean("atlas.authentication.method.kafka.message.principal", false);
+
+            if(msgPrincipalEnabled) {
+                if (!StringUtils.isEmpty(messageUser)) {
+                    List<GrantedAuthority> grantedAuths = getAuthoritiesFromUGI(messageUser);
+                    UserDetails principal;
+
+                    if (grantedAuths.isEmpty()) {
+                        principal = new User(messageUser, "", new HashSet<>());
+                    } else {
+                        principal = new User(messageUser, "", grantedAuths);
+                    }
+
+                    Authentication finalAuthentication = new UsernamePasswordAuthenticationToken(principal, "");
+                    SecurityContextHolder.getContext().setAuthentication(finalAuthentication);
```

Comment:
This will be adding authentication in context for every message and this can cause performance issue. Can we maintain session some how ?

Reply:
I think it is difficult to say about the session here, since every kafka message processed can potentially have different user or properties. 
I see it more similar in nature to REST call, where you authenticate every request.